native/src/main/java/io/ballerina/stdlib/graphql/runtime/engine/ExecutorVisitor.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.stdlib.graphql.runtime.engine;
+
+import io.ballerina.runtime.api.creators.ValueCreator;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BObject;
+import io.ballerina.runtime.api.values.BString;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+import static io.ballerina.stdlib.graphql.runtime.utils.ModuleUtils.getModule;
+
+/**
+ * This class provides native implementations of the Ballerina ExecutorVisitor class.
+ */
+public class ExecutorVisitor {
+    private static final String DATA_MAP = "dataMap";
+    private static final String DATA_RECORD_NAME = "Data";
+    private static final Object NullObject = new Object();
+    private final ConcurrentHashMap<BString, Object> dataMap = new ConcurrentHashMap<>();
```

Comment:
When initializing the ConcurrentHashMap, we can configure the concurrency level.

* `concurrencyLevel`: The estimated number of concurrently updating threads. The implementation may use this value as a sizing hint. (Default: 16)

Is it okay to use the default value, or do we need to provide a way to configure it? WDYT @ThisaruGuruge ?

Reply:
AFAIK, the `concurrencyLevel` is not significant in the later Java versions. So, let's not worry about this for now. We can improve this later if we face any issues.